byron/ledger/executable-spec/test/Ledger/Delegation/Properties.hs

```diff
@@ -167,16 +176,22 @@ instance STS DBLOCK where
 
   initialRules
     = [ do
-          IRC (env) <- judgmentContext
+          IRC env <- judgmentContext
           pure (env, initialDIState)
       ]
 
   transitionRules
     = [ do
           TRC (_, (env, st), dblock) <- judgmentContext
-          env ^. slot < dblock ^. blockSlot ?! NotIncreasingBlockSlot
+          let nextSlot = dblock ^. blockSlot
+          env ^. slot < nextSlot ?! NotIncreasingBlockSlot
           stNext <- trans @DELEG $ TRC (env, st, dblock ^. blockCerts)
-          return (env & slot .~ dblock ^. blockSlot, stNext)
+          -- We fix the number of slots per epoch to 10. This is the same
+          -- constant used in production.
+          let nextEpoch = Epoch $ unSlot nextSlot `div` 10
```

Comment:
Shouldn't the `div 10` part be obtained by a function from a logic available elsewhere? This way it seems rather arbitrary.

Reply:
Yeah. What I wrote is actually wrong. But I'm gonna introduce a function to calculate the number of slots per epoch given the `k` parameter.